### DISCUSSION TITLE
Deleting wiki pages

### DISCUSSION THREAD

**Discussion by guyemerson** [delph-in/docs#45] [General]
Something I promised to look into after the summit was displaying a list of deleted pages on the wiki.  As far as I can tell, GitHub doesn't provide any way to automatically display deleted pages.  The wiki is internally stored as a git repo, so it's possible to clone the wiki and retrieve a list of deleted pages, but that would be an inconvenience.

One option would be to set up a wiki page which is a list of deleted pages.  We might be able to set this up to automatically update, perhaps using [Git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) or [GitHub actions](https://docs.github.com/en/actions), but this might be overkill, giving us an extra thing to maintain.

If the main use case is deleting pages that have become obsolete (so that they don't show up in search results and confuse people), I wonder if it would be simpler and easier to indicate this on a per-page basis.  For instance, instead of deleting a page completely, its content could be replaced by a generic message like "This page has become obsolete.", optionally giving links to more up-to-date pages.

**Comment by emilymbender** [comment]
If there's a manual step involved each time, then having a template that we can paste in ("This page has become obsolete" or similar) seems like it wouldn't be more effort up front and it would capture what we want.

**Reply by arademaker** [reply]
I didn't understand the motivation. The pages I removed after migration were only duplications (cases by the migration process and some unicodes in the file names). 

We still have some open issues regarding the errors on the migration. I hope to find some time to finish the issues.

The command "git whatchanged" can easily report deleted files whenever one suspect something relevant were removed, why duplicate that information? 

Unfortunately, the delete action in the web interface of GitHub wiki do not have an option to write a message. If we delete locally in a clone from the git, we can write commit messages. So maybe using that approach can be better for the future instead of deleting in the web interface.

**Reply by guyemerson** [reply]
To clarify, this isn't about problems with migration!  (Thanks again for all your work on that!)

Yes, git commands can be used to get any information about deleted files, but this requires cloning the repo (and understanding the necessary commands).

One motivation is old documentation pages which might be referenced somewhere.  If the page is obsolete, we don't want to keep the page in case it confuses people.  But if we delete the page and it's still referenced somewhere, it could also confuse someone who is specifically looking for it and can't find it (or rather, can't find it without cloning the wiki repo).  So to balance those two needs, it would be useful to delete the page, but have a clear indication that the page was deleted.

**Reply by arademaker** [reply]
I see. So the use case is really to avoid the creation of broken links? Interesting problem. Maybe we can find a way to add this check or do it periodically somehow. 

Indeed another functionality that MoinMoin has and GitHub wiki doesn't. On the other hand, I believe it may be time to think in an alternative documentation approach instead of wiki.

**Reply by guyemerson** [reply]
If there are links from outside the wiki (e.g. in a paper, in a thesis, from the Discourse site, etc.), then we can't easily check for them.

We'll have to deal with this problem, no matter what platform we use or what approach we take to documentation.
